### DISCUSSION TITLE
Aberration free lens model

### DISCUSSION THREAD

**Discussion by osepsi** [mjhoptics/ray-optics#139] [Q&A]
In my practice, I often find myself in the situation that I need to introduce an objective lens into the optical system where no data of the actual elements can be found from the producer, only the standard spec sheet parameters. In these cases I usually model the objective lens as a paraxial lens, that is it is assumed that the objective lens can be taken as aberration free optical element.

Unfortunately in the ray-optics code I have not found the proper solution to introduce an aberration free lens. I have tried the ThinLens object, but it introduces aberrations (this is normal, I know). Is there a way to somehow simply introduce aberration free lens in this package?

**Comment by mjhoptics** [comment]
Hello @osepsi,
I understand what you want and why you want it, but the laws of geometrical optics, on which the `ray-optics` ray trace is based, prohibit such behavior - it isn't achievable even under very general assumptions (see Hamiltonian optics). You can have optical systems which image perfectly at a single pair of point conjugates but not at a second pair. Consider an off-axis object point: it's image space conjugate forms a second set of conjugate points but cannot image perfectly. Curved field is an image defect that can't be worked around when using a non-paraxial (aka real, trigonometric, Snell's Law) ray trace.
If you truly need to model part of a system, say the eyepiece in a telescope design, start from intermediate image planes rather than the system object plane. If you know you're using an objective with 1m fl but don't have a prescription, omit the objective from the model and make the object plane with the specs of the objective's image plane. In many cases this will be a virtual object for the system you're trying to model; furthermore, the entrance pupil would be located where the objective would be. Explicitly make surface 1 a dummy plane at the entrance pupil position, then start your system prescription at surface 2.
If there's another aspect to your problem I don't appreciate, please respond and clarify for me.
Thanks
Mike Hayford
